src/components/implementation/no_interface/llbooter/llbooter.c

```diff
@@ -348,14 +427,50 @@ void
 init_done(int parallel_init, init_main_t main_type)
 {
 	compid_t client = (compid_t)cos_inv_token();
-	struct crt_comp *c;
+	struct crt_comp    *c;
+	struct crt_chkpt   *chkpt;
+	struct crt_comp    *new_comp  = boot_comp_get(crt_ncomp() + 1);
+	thdcap_t 		thdcap;
+	int 			ret;
+	char			*name;
 
 	assert(client > 0 && client <= MAX_NUM_COMPS);
 	c = boot_comp_get(client);
 
 	crt_compinit_done(c, parallel_init, main_type);
 
+#ifdef ENABLE_CHKPT
+	if (c->id == cos_compid()) {
+	 	/* don't allow chkpnts of the booter */
+	 	BUG();
+	}
+
+	name = "chkpt_";
+	strncat(name, c->name, strlen(c->name));
```

Comment:
Isn't this a bug? name doesn't have any space allocated after the static string for `c->name`. So you're just writing into RO memory indiscriminately.

When you do properly allocate `name` (statically in the function is fine), you'll want to add an assertion that `strlen(c->name) + strlen("chkpt_") < whatever the size of name is`?